tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/debugger_container.ts

```diff
@@ -12,20 +12,39 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-import {Component} from '@angular/core';
-import {Store} from '@ngrx/store';
+import {Component, OnInit} from '@angular/core';
+import {createSelector, select, Store} from '@ngrx/store';
+import {DebuggerRunListing, State} from './store/debugger_types';
 
-/** @typehack */ import * as _typeHackRxjs from 'rxjs';
+import {debuggerLoaded} from './actions';
+import {getDebuggerRuns} from './store';
 
-// TODO(cais): Move to a separate file.
-export interface State {}
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
 
 @Component({
   selector: 'tf-debugger-v2',
   template: `
-    <debugger-component></debugger-component>
+    <debugger-component
+      [runs]="runs$ | async"
+      [runIds]="runsIds$ | async"
+    ></debugger-component>
   `,
 })
-export class DebuggerContainer {
+export class DebuggerContainer implements OnInit {
+  readonly runs$ = this.store.pipe(select(getDebuggerRuns));
+
+  readonly runsIds$ = this.store.pipe(
+    select(
+      createSelector(
+        getDebuggerRuns,
+        (runs): string[] => Object.keys(runs)
+      )
+    )
+  );
+
   constructor(private readonly store: Store<State>) {}
+
+  ngOnInit(): void {
+    this.store.dispatch(debuggerLoaded());
```

Comment:
Optional but consider this. There are rare cases when you want to use this. We should avoid that certain view is initialized if possible since that can lead to proliferation of bad patterns. Since view is a reflection of state in the store, effects and reducers should be able to update correctly without this. 

For instance, whatever you do here can probably be replaced with PluginChanged action, right?

Reply:
> Also, why do you say these are rare cases? The DebuggerContainer being loaded

It is rare because this information is easily attainable from store's perspective and view really should not fire an action with intention of telling the store. 

> which is at a higher level than the individual plugin (DebuggerV2 specifically), should know something about the internal workings of the plugin.

I am not sure if I agree with this. One of the point of building on the same framework and same instance of the store is better "integration". Being part of the greater system is not a wrong thing.

Reply:
I added a TODO item here to explore consolidate this effect into the integrated webapp.